src/pages/index.js

```diff
@@ -41,38 +59,77 @@ function handleImageClick(name, link) {
 }
 
 //Popup Forms
-const profilePopup = new PopupWithForm(profileModal, (data) => {
-  console.log(data);
+const profilePopup = new PopupWithForm(profileModal, (data, button) => {
   userInfo.setUserInfo(data);
-  profilePopup.closePopup();
-});
-const cardForm = new PopupWithForm(newCardModal, (data) => {
-  const name = newCardTitle.value;
-  const link = newCardURL.value;
-  addCard({ name, link }, mediaList);
-  cardForm.closePopup();
+  api.saveUserInfo(data).then(() => {
+    button.textContent = "Save";
+    profilePopup.closePopup();
+  });
+  button.textContent = "Saving...";
 });
+
+const profilePicturePopup = new PopupWithForm(
+  profilePictureModal,
+  (data, button) => {
+    api
+      .updatePicture(data.url)
+      .then((res) => {
+        profileAvatar.src = data.url;
+        button.textContent = "Save";
+        profilePicturePopup.closePopup();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+    button.textContent = "Saving...";
+  }
+);
+const deletePopup = new PopupWithFormSubmit(deleteModal);
+deletePopup.setEventListeners();
+
+function handleDeleteCard(cardId, cardElement) {
+  const id = cardId;
+  deletePopup.openPopup();
+  deletePopup.setSubmitAction(() => {
+    api.deleteCard(id);
+    cardElement.remove();
```

Comment:
You need to **delete** cards only in block `then` after a **successful** response from the server so that if there is an error the card would stay in the `DOM`.